sdl2promela/promela/modelbuilder.py

```diff
@@ -368,6 +369,105 @@ def withRight(self, right: Statement) -> "BinaryExpressionBuilder":
         return self
 
 
+class UnaryExpressionBuilder:
+    """Unary expression builder."""
+
+    expression: UnaryExpression
+    """Build unary expression."""
+
+    def __init__(self, operator: UnaryOperator):
+        self.expression = UnaryExpression()
+        self.expression.operator = operator
+
+    def build(self) -> UnaryExpression:
+        """
+        Retrieve the built unary expression.
+        :returns: The built expression.
+        """
+        return self.expression
+
+    def withExpression(self, expr: Expression) -> "UnaryExpressionBuilder":
+        """
+        Set the expression.
+        :param expr: the expression to be set.
+        :returns: The buider itself (for call chaining).
+        """
+        self.expression.expr = expr
+        return self
+
+
+class ArrayAccessBuilder:
+    """ArrayAccess builder."""
+
+    arrayAccess: ArrayAccess
+    """Built variable reference."""
+
+    def __init__(self):
+        self.arrayAccess = ArrayAccess()
+
+    def build(self) -> ArrayAccess:
+        """
+        Retrieve the built array access expression.
+        :returns: The built expression.
+        """
+        return self.arrayAccess
+
+    def withArray(
+        self, array: Union[VariableReference, MtypeAccess]
```

Comment:
This is a bit confusing. Maybe a better name?

Reply:
Changed to MemberAccess